simapp/v2/simdv2/cmd/root_di.go

```diff
@@ -1,161 +1,101 @@
-	autoCliOpts.ModuleOptions = make(map[string]*autocliv1.ModuleOptions)
-	autoCliOpts.ModuleOptions[nodeCmds.Name()] = nodeCmds.AutoCLIOptions()
-	if err := autoCliOpts.EnhanceRootCommand(rootCmd); err != nil {
-		panic(err)
+	var (
+		autoCliOpts     autocli.AppOptions
+		moduleManager   *runtime.MM[T]
+		clientCtx       client.Context
+		simApp          *simapp.SimApp[T]
+		depinjectConfig = depinject.Configs(
+			depinject.Configs(
+				depinject.Supply(logger, runtime.GlobalConfig(configMap)),
+				depinject.Provide(ProvideClientContext),
+			),
+		)
+	)
+	if isAppRequired(subCommand) {
+		// server construction
+		simApp, err = simapp.NewSimApp[T](depinjectConfig, &autoCliOpts, &moduleManager, &clientCtx)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		// client construction
+		if err = depinject.Inject(
+			depinject.Configs(
+				simapp.AppConfig(),
+				depinjectConfig,
+			),
+			&autoCliOpts, &moduleManager, &clientCtx,
+		); err != nil {
+			return nil, err
+		}
 	}
 
-	return rootCmd
-}
-
-func ProvideClientContext(
-	appCodec codec.Codec,
-	interfaceRegistry codectypes.InterfaceRegistry,
-	txConfigOpts tx.ConfigOptions,
-	legacyAmino registry.AminoRegistrar,
-	addressCodec address.Codec,
-	validatorAddressCodec address.ValidatorAddressCodec,
-	consensusAddressCodec address.ConsensusAddressCodec,
-) client.Context {
-	var err error
-
-	amino, ok := legacyAmino.(*codec.LegacyAmino)
-	if !ok {
-		panic("registry.AminoRegistrar must be an *codec.LegacyAmino instance for legacy ClientContext")
+	commandDeps := commandDependencies[T]{
+		globalAppConfig: configMap,
+		txConfig:        clientCtx.TxConfig,
+		moduleManager:   moduleManager,
+		simApp:          simApp,
 	}
-
-	clientCtx := client.Context{}.
-		WithCodec(appCodec).
-		WithInterfaceRegistry(interfaceRegistry).
-		WithLegacyAmino(amino).
-		WithInput(os.Stdin).
-		WithAccountRetriever(types.AccountRetriever{}).
-		WithAddressCodec(addressCodec).
-		WithValidatorAddressCodec(validatorAddressCodec).
-		WithConsensusAddressCodec(consensusAddressCodec).
-		WithHomeDir(simapp.DefaultNodeHome).
-		WithViper("") // uses by default the binary name as prefix
-
-	// Read the config to overwrite the default values with the values from the config file
-	customClientTemplate, customClientConfig := initClientConfig()
-	clientCtx, err = config.CreateClientConfig(clientCtx, customClientTemplate, customClientConfig)
-	if err != nil {
-		panic(err)
+	rootCommand = &cobra.Command{
+		Use:               "simdv2",
+		Short:             "simulation app",
+		SilenceErrors:     true,
+		PersistentPreRunE: rootCommandPersistentPreRun(clientCtx),
 	}
-
-	// textual is enabled by default, we need to re-create the tx config grpc instead of bank keeper.
-	txConfigOpts.TextualCoinMetadataQueryFn = authtxconfig.NewGRPCCoinMetadataQueryFn(clientCtx)
-	txConfig, err := tx.NewTxConfigWithOptions(clientCtx.Codec, txConfigOpts)
+	factory.EnhanceCommand(rootCommand)
```

Comment:
Nice! Consistency between server/v2 and client/v2 api is great 👌🏾

Maybe we should rename to EnhanceRootCommand so that the APIs are named the same?